chain/ethereum/src/network_indexer/network_indexer.rs

```diff
@@ -0,0 +1,1196 @@
+                        ),
+                    );
+
+                    // Chain head wasn't new, try getting a new one.
+                    transition!(PollChainHead {
+                        local_head: state.local_head,
+                        prev_chain_head: Some(chain_head.into()),
+                        chain_head: poll_chain_head(context),
+                    });
+                }
+
+                // Ignore the chain head if its number is below the current local head;
+                // it would mean we're switching to a shorter chain, which makes no sense
+                if chain_head.number.unwrap().as_u64()
+                    < state.local_head.map_or(0u64, |ptr| ptr.number)
+                {
+                    debug!(
+                        context.logger,
+                        "Chain head is for a shorter chain; poll chain head again";
+                        "local_head" => state.local_head.map_or(
+                            String::from("none"), |ptr| format!("{}", ptr)
+                        ),
+                        "chain_head" => chain_head.format(),
+                    );
+
+                    transition!(PollChainHead {
+                        local_head: state.local_head,
+                        prev_chain_head: state.prev_chain_head,
+                        chain_head: poll_chain_head(context),
+                    });
+                }
+
+                // Calculate the number of blocks remaining before we are in sync with the
+                // network; fetch no more than 1000 blocks at a time.
+                let chain_head_number = chain_head.number.unwrap().as_u64();
+                let next_block_number = state.local_head.map_or(0u64, |ptr| ptr.number + 1);
+                let remaining_blocks = chain_head_number + 1 - next_block_number;
+                let block_range_size = remaining_blocks.min(1000);
+                let block_numbers = next_block_number..(next_block_number + block_range_size);
+
+                // Ensure we're not trying to fetch beyond the current chain head (note: the
+                // block numbers range end is _exclusive_, hence it must not be greater than
+                // chain head + 1)
+                assert!(
+                    block_numbers.end <= chain_head_number + 1,
+                    "overfetching beyond the chain head; \
+                     this is a bug in the block range calculation"
+                );
+
+                info!(
+                    context.logger,
+                    "Process {} of {} remaining blocks",
+                    block_range_size, remaining_blocks;
+                    "chain_head" => format!("{}", chain_head.format()),
+                    "local_head" => state.local_head.map_or(
+                        String::from("none"), |ptr| format!("{}", ptr)
+                    ),
+                    "range" => format!("#{}..#{}", block_numbers.start, block_numbers.end-1),
```

Comment:
Lets put the range in this log in square brackets to clarify that it's inclusive on both ends.

Reply:
Done